h2o-automl/src/main/java/water/automl/api/schemas3/AutoMLBuildSpecV99.java

```diff
@@ -118,16 +119,51 @@ public AutoMLBuildControlV99() {
 
   } // class AutoMLInputV99
 
+  public static final class AutoMLStoppingCriteriaV99 extends Schema<AutoMLStoppingCriteria, AutoMLStoppingCriteriaV99> {
+
+    @API(help = "Seed for random number generator; set to a value other than -1 for reproducibility.", direction = API.Direction.INOUT)
+    public long seed;
+
+    @API(help = "Maximum number of models to build (optional).", direction = API.Direction.INOUT)
+    public int max_models;
+
+    @API(help = "Maximum time to spend building all models (optional).", direction = API.Direction.INOUT)
+    public double max_runtime_secs;
+
+    @API(help = "Maximum time to spend on each individual model (optional).", direction = API.Direction.INOUT)
+    public double max_runtime_secs_per_model;
+
+    @API(help = "Early stopping based on convergence of stopping_metric. Stop if simple moving average of length k of the stopping_metric does not improve for k:=stopping_rounds scoring events (0 to disable)", level = API.Level.secondary, direction=API.Direction.INOUT)
+    public int stopping_rounds;
+
+    @API(help = "Metric to use for early stopping (AUTO: logloss for classification, deviance for regression)", valuesProvider = StoppingMetricValuesProvider.class, level = API.Level.secondary, direction=API.Direction.INOUT)
+    public ScoreKeeper.StoppingMetric stopping_metric;
+
+    @API(help = "Relative tolerance for metric-based stopping criterion (stop if relative improvement is not at least this much)", level = API.Level.secondary, direction=API.Direction.INOUT)
```

Comment:
@seb-h2o here I'm just curious how we decide when level should be set to secondary ?

Reply:
didn't notice this `@API` param until now, thanks for pointing at it: I'm afraid it is not used consistently currently:
allowed values are:
`critical` : seems to be a duplicate for the `required` param, and this is the default value... maybe this is the reason why this is ignored, as `required` is doing a better job.
`secondary`: probably relevant for many params.
`expert`: the difference with `secondary` is getting obscure now...
Looks like a not-that-useful param.

Reply:
after short investigation, seems to be used to group params in Flow

Reply:
@seb-h2o this is actually what I'd initially thought about these levels. Then back to the question how to decide which one to use. Should we find a guideline for that?